dp/drivers/i40e.c

```diff
@@ -354,9 +354,30 @@ static void promiscuous_disable(struct ix_rte_eth_dev *dev)
 	rte_eth_promiscuous_disable(dev->port);
 }
 
+static DEFINE_SPINLOCK(i40e_dev_lock);
+
 static int reta_update(struct ix_rte_eth_dev *dev, struct rte_eth_rss_reta *reta_conf)
 {
-	panic("reta update not implemented\n");
+	int ret, i;
+	struct rte_eth_rss_reta_entry64 r_reta_conf[2];
+	struct rte_eth_dev r_dev = rte_eth_devices[dev->port];
+
+	/* first convert reta_conf to dpdk format*/
+	r_reta_conf[0].mask = reta_conf->mask_lo;
+	r_reta_conf[1].mask = reta_conf->mask_hi;
+
+	for (i = 0; i < RTE_RETA_GROUP_SIZE; ++i) {
+		r_reta_conf[0].reta[i] = reta_conf->reta[i];
+		r_reta_conf[1].reta[i] = reta_conf->reta[RTE_RETA_GROUP_SIZE + i];
+	}
+
+	spin_lock(&i40e_dev_lock);
+	ret = r_dev.dev_ops->reta_update(&r_dev, r_reta_conf, ETH_RSS_RETA_NUM_ENTRIES);
```

Comment:
ETH_RSS_RETA_NUM_ENTRIES would need to be dynamically changed depending on the driver (ixgbe or i40e).
https://github.com/ix-project/ix/blob/bd4dd0ba200413716071561a3c3f2994159c665d/inc/ix/ethdev.h#L225